crates/core/tedge_agent/src/agent.rs

```diff
@@ -555,13 +555,22 @@ impl SmAgent {
             .await?;
         let () = restart_operation::create_slash_run_file(&self.config.run_dir)?;
 
-        let _process_result = std::process::Command::new("sudo").arg("sync").status();
         // state = "Restarting"
-        match std::process::Command::new("sudo")
-            .arg(INIT_COMMAND)
-            .arg("6")
-            .status()
-        {
+        let mut command = if cfg!(test) {
+            // running `echo 6` with no sudo
+            let mut test_command = std::process::Command::new(INIT_COMMAND);
+            test_command.arg("6");
+            test_command
+        } else {
+            // sync first
+            let _process_result = std::process::Command::new("sudo").arg("sync").status();
+            // running `sudo init 6`
+            let mut true_command = std::process::Command::new("sudo");
+            true_command.arg(INIT_COMMAND).arg("6");
+            true_command
+        };
+
+        match command.status() {
```

Comment:
@didier-wenzek please take a look at this. I made this change here so that we can execute the `check_agent_restart_file_is_created` test - which was previously ignored. After some investigating I found out that it was ignored because it was running `sudo sync` even in the test.

Reply:
This needs to be done sure. However, rather than using `if cfg!(test)`, I would use the same pattern as for defining the INIT_COMMAND`.

* If you look this code it's not obvious that the `INIT_COMMAND` line 561 is not the same as for line 569.
* So why not an `init_command()` function with two implementations, one for `#[cfg(not(test))]` and another one for `#[cfg(test)]`.